lib/api/core/restrictedStorage.js

```diff
@@ -21,22 +21,39 @@
 
 'use strict';
 
-// @todo maybe rename to RestrictedStorage because it's limited to one index
+const Bluebird = require('bluebird');
 
 /**
  * Wrapper around a storage engine.
  * Once instantiated, this class can only access the index passed in the
  * constructor
  */
-class InternalStorage {
+class RestrictedStorage {
   /**
    *
+   * @param {Kuzzle} kuzzle
    * @param {String} index
-   * @param {StorageEngin} storageEngine
+   * @param {StorageEngine} storageEngine
    */
-  constructor (index, storageEngine) {
+  constructor (kuzzle, index, storageEngine) {
+    this.kuzzle = kuzzle,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove use of this comma operator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS878)